dftimewolf/lib/collectors/aws.py

```diff
@@ -0,0 +1,255 @@
+        self.remote_zone, aws_profile=self.remote_profile_name)
+
+    self.incident_id = incident_id
+    self.remote_instance_id = remote_instance_id
+
+    self.volume_ids = volume_ids.split(',') if volume_ids else []
+    self.all_volumes = all_volumes
+    self.analysis_zone = analysis_zone or remote_zone
+    self.analysis_profile_name = analysis_profile_name or remote_profile_name
+
+    # See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html
+    self.device_suffixes = list('fghijklmnop')
+
+    analysis_vm_name = 'aws-forensics-vm-{0:s}'.format(self.incident_id)
+    print('Your analysis VM will be: {0:s}'.format(analysis_vm_name))
+    self.state.StoreContainer(
+        containers.TicketAttribute(
+            name=self._ANALYSIS_VM_CONTAINER_ATTRIBUTE_NAME,
+            type_=self._ANALYSIS_VM_CONTAINER_ATTRIBUTE_TYPE,
+            value=analysis_vm_name))
+    self.analysis_vm, _ = aws_forensics.StartAnalysisVm(
+        analysis_vm_name,
+        self.analysis_zone,
+        boot_volume_size,
+        ami=ami,
+        cpu_cores=cpu_cores,
+        dst_profile=self.analysis_profile_name,
+    )
+
+  def _GetVolumesFromIds(self, volume_ids):
+    """Gets volumes from an account by volume IDs.
+
+    Args:
+      volume_ids (list[str]): List of volume ids to get from the account.
+
+    Returns:
+      list[AWSVolume]: List of AWSVolume objects to copy.
+    """
+    volumes = []
+    for volume_id in volume_ids:
+      try:
+        volumes.append(self.source_account.GetVolumeById(volume_id))
+      except RuntimeError:
+        self.state.AddError(
+            'Volume "{0:s}" was not found in AWS account {1:s}'.format(
+                volume_id, self.remote_profile_name),
+            critical=True)
+        return []
+    return volumes
+
+  def _GetVolumesFromInstance(self, instance_id, all_volumes):
+    """Gets volumes to copy based on an instance name.
+
+    Args:
+      instance_id (str): Instance ID of the instance to get the volumes from.
+      all_volumes (bool): If True, get all volumes attached to the instance. If
+          False, get only the instance's boot volume.
+
+    Returns:
+      list[AWSVolume]: List of AWSVolume objects to copy.
+    """
+    try:
+      remote_instance = self.source_account.GetInstanceById(instance_id)
+    except RuntimeError as exception:
+      self.state.AddError(str(exception), critical=True)
+      return []
+
+    if all_volumes:
+      return list(remote_instance.ListVolumes().values())
+    return [remote_instance.GetBootVolume()]
+
+  def _FindVolumesToCopy(self):
+    """Determines which volumes to copy depending on the collector's attributes.
+
+    Returns:
+      list[AWSVolume]: A list of the volumes to copy.
+    """
+    if not (self.remote_instance_id or self.volume_ids):
+      self.state.AddError(
+          'You need to specify at least an instance name or volume ids to copy',
+          critical=True)
+      return []
```

Comment:
This should go in `SetUp`. We want to be able to fail early if these parameters are not provided. (Should also be changed in `gcloud.py`, #297 )

Reply:
Thanks, opened #298 for `gcloud.py`